src/plugins/topic_viewer/TopicViewer.qml

```diff
@@ -0,0 +1,252 @@
+        function itemData ()
+        {
+            return {
+                "name": model.name,
+                "type": model.type,
+                "path": model.path,
+                "topic": model.topic
+            }
+        }
+
+        MouseArea {
+            id: dragMouse
+            anchors.fill: parent
+
+            // only plottable items are dragable
+            drag.target: (model === null) ? null :
+                         (model.plottable) ? parent : null
+
+            // get a copy image of the dragged item
+            onPressed: parent.grabToImage(function(result) {
+                parent.Drag.imageSource = result.url
+            })
+
+            onReleased:
+            {
+                // emit drop event to notify the DropArea (must manually)
+                parent.Drag.drop();
+            }
+
+            hoverEnabled: true
+            propagateComposedEvents: true
+            // make the cursor with a drag shape at the plottable items
+            cursorShape: (model === null) ?  Qt.ArrowCursor : (model.plottable) ?
+                                                Qt.DragCopyCursor : Qt.ArrowCursor
+
+            onClicked: {
+                // change the selection of the tree by clearing the prev, select a new one
+                tree.selection.select(styleData.index,ItemSelectionModel.ClearAndSelect)
+
+                // set the selection index to the index of the clicked item (must set manually)
+                tree.selection.setCurrentIndex(styleData.index,ItemSelectionModel.ClearAndSelect)
+
+                // the currentIndex of the tree.selection is not the same
+                // of the tree.currentIndex, so set the tree.currentIndex.
+                // this is the way to access it as it is read-only
+                tree.__currentRow = styleData.row
+
+                // set the focus to the selected item to receive the keyboard events
+                // this is useful to enable navigating with keyboard from the right position
+                item.forceActiveFocus();
+
+                tree.expandCollapseMsg(tree.currentIndex);
+            }
+        }
+
+        Image {
+            id: icon
+            source: "plottable_icon.svg"
+            height: itemHeight * 0.6
+            width: itemHeight * 0.6
+            y : itemHeight * 0.2
+            visible: (model === null) ? false : model.plottable
+        }
+
+        Text {
+            id : field
+            text: (model === null) ? "" : model.name
+            color: (Material.theme == Material.Light || styleData.selected) ?
+                    Material.color(Material.Grey, Material.Shade800):
+                    Material.color(Material.Grey, Material.Shade400);
+
+            font.pointSize: 12
+            anchors.leftMargin: 5
+            anchors.left: icon.right
+            y: icon.y
+        }
+
+        ToolTip {
+            id: tool_tip
+            delay: 1000
```

Comment:
The animation is quite slow right now :sweat_smile: How about making it much quicker:

```suggestion
            delay: 100
```